README.md

```diff
@@ -74,6 +74,7 @@ Details:
    * `verify` attribute - if set to false, disables verification of the resulting CAP file with offcardeverifier. Optional.
    * `debug` attribute - if set to true, generates debug CAP components. Optional.
    * `ints` attribute - if set to true, enables support for 32 bit `int` type. Optional.
+   * `javaVersion` attribute - if set to true, override the Java source and target version. Optional.
```

Comment:
I think I missed this. What is the reasoning for this? Correct me if wrong, but the maximum Java version is limited by JC kit. Why would one want to change this to *lower*?

Also, the styling for names is all lower case, in this case could be "java".

Reply:
The actual java version is only limited in practice by some of the oldest (JC 2.1) kits, at least not beyond the "1.5" level that I tend to use. This practice comes from a project where we used static verification tools based on compile-time-only annotations. An openly available example is checkstyle, not that I checked that one can actually use that.

So I do not actually lower the level, I raise it to 1.5. Doing so is a bit of a hack but useful because of said verification tools. Most if not all non-usable Java 1.5 features will still fail the build for other reasons, but compile-time annotations will not.

Lowering the level is indeed quite useless. Specifically the features added before 1.5 are of no use at all to JC (strictfp in 1.2, assert in 1.4). Features in 1.5 are mostly unusable without appropriate runtime support. Certain checking annotations (@Override) from 1.5 just work on JC-2.2 and up.

Its a hack, what can I say. I am willing to drop this feature and practice for my public projects if it puts you off since none of those actually depend on this seriously.

Maybe there should rather be a feature to pass arbitrary arguments to the compiler, although that won't actually work in this case.

I can change the parameter name of course. Indeed lowercase seems to be general practice in ant.

Reply:
ok. Lets have it as "javaversion" ? And the README line should probably not read "if set to true".